vault/main.tf

```diff
@@ -0,0 +1,55 @@
+terraform {
+  required_version = ">= 1.0"
+
+  required_providers {
+    coder = {
+      source  = "coder/coder"
+      version = ">= 0.12"
+    }
+  }
+}
+
+# Add required variables for your modules and remove any unneeded variables
+variable "agent_id" {
+  type        = string
+  description = "The ID of a Coder agent."
+}
+
+variable "vault_addr" {
+  type        = string
+  description = "The address of the Vault server."
+}
+
+variable "vault_auth_id" {
+  type        = string
+  description = "The ID of the Vault auth method to use."
+  default     = "vault"
+}
+
+variable "vault_cli_version" {
+  type        = string
+  description = "The version of Vault to install."
+  default     = "latest"
+  # validate the version is in the format x.y.z or latest
+  validation {
+    condition     = can(regex("^(latest|[0-9]+\\.[0-9]+\\.[0-9]+)$", var.vault_cli_version))
+    error_message = "Vault version must be in the format 0.0.0 or latest"
+  }
+}
+
+resource "coder_script" "vault" {
+  agent_id     = var.agent_id
+  display_name = "vault"
+  icon         = "/icon/vault.svg"
```

Comment:
Have you tested this icon? Those outside coder/coder/site/static haven't rendered in my tests.

Reply:
İ have a PR to add to there.

Reply:
It should render now PR is merged.

Reply:
Great. If you want to add tests before merging go ahead, or I can add later.

Reply:
Sure. I am going to merge it without tests. İt would be great if you can add tests.